bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/observability/tracer/TracersStore.java

```diff
@@ -51,31 +52,29 @@ private TracersStore() {
     public void loadTracers() {
         ConfigRegistry configRegistry = ConfigRegistry.getInstance();
         if (configRegistry.getAsBoolean(CONFIG_TRACING_ENABLED)) {
-
-            this.tracerStore = new HashMap<>();
+            String overallProviderName = configRegistry.getConfigOrDefault(CONFIG_OBSERVABILITY_PROVIDER, JAEGER);
+            String tracerName = configRegistry.getConfigOrDefault(TRACER_NAME_CONFIG, overallProviderName);
 
             ServiceLoader<OpenTracer> openTracers = ServiceLoader.load(OpenTracer.class);
-            HashMap<String, OpenTracer> tracerMap = new HashMap<>();
-            openTracers.forEach(t -> tracerMap.put(t.getName().toLowerCase(), t));
-
-            String defaultReporterName = configRegistry.getConfigOrDefault(CONFIG_OBSERVABILITY_PROVIDER, JAEGER);
-            String tracerName = configRegistry.getConfigOrDefault(TRACER_NAME_CONFIG, defaultReporterName);
+            for (OpenTracer openTracer : openTracers) {
+                if (Objects.equals(openTracer.getName().toLowerCase(), tracerName.toLowerCase())) {
```

Comment:
can't we use `tracerName.equalsIgnoreCase(openTracer.getName())`

Reply:
Fixed